### PR TITLE
feat(common): new pipe to support proper case conversion

### PR DESCRIPTION
**Pull request by AlmogShaul**
Please check if the PR fulfills these requirements

The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
Tests for the changes have been added (for bug fixes / features)
Docs have been added / updated (for bug fixes / features)
What kind of change does this PR introduce? (check one with "x")

[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
What is the current behavior? (You can also link to an open issue here)

What is the new behavior?

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No
If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

Other information:

Add propercase_pipe which convert string to propercase

'hello there'->'Hello there'

### CODE REVIEW

**Comment by vicb** [comment]
- Please merge the commits,
- I think it would make sense to move all case conversion pipes in a single file

File: modules/@angular/common/src/pipes/propercase_pipe.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * @stable
+ */
+@Pipe({name: 'propercase'})
+export class ProperCasePipe implements PipeTransform {
+  transform(value: string): string {
+    if (isBlank(value)) return value;
+    if (!isString(value)) {
+      throw new InvalidPipeArgumentError(ProperCasePipe, value);
+    }
+    return value[0].toUpperCase() + value.substr(1);
```
Comment (metanav) [inline modules/@angular/common/src/pipes/propercase_pipe.ts]: I think 'value' should be converted to lowercase first to cover all cases:
 'hello there or HELLO THERE or hElLo tHeRe' will be transformed to 'Hello there'.

value = value.toLowerCase();
return value[0].toUpperCase() + value.substr(1);

Comment (AlmogShaul) [inline modules/@angular/common/src/pipes/propercase_pipe.ts]: right,I will fix it

Comment (AlmogShaul) [inline modules/@angular/common/src/pipes/propercase_pipe.ts]: @vicb To make a single file with all the case conversion pipes, or to wait with that?

**Comment by AlmogShaul** [comment]
@metanav I need help squashing my pushed commits.

**Comment by vicb** [comment]
1 - merge all case conversion pipes,
2 - squash the commits,
3 - the resulting commit comment should stick the conventions: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format 

To squash commits:
```
git checkout master
git fetch
git merge origin/master --ff

git checkout camelCasePipe
git rebase master
git rebase matser -i  # pick the first commit, and replace pick with squash for following ones
```

(you'll find plenty of help in github... or in google)

**Comment by AlmogShaul** [comment]
I authored these commits

**Pull request closed without merging**